tests/production_pause_max_rev_blks_test.py

```diff
@@ -0,0 +1,166 @@
+import os
+import shutil
+import signal
+import time
+
+from TestHarness import Cluster, TestHelper, Utils, WalletMgr
+
+####################################################################################
+# production_pause_max_rev_blks_test
+# Test --max-reversible-blocks works as expected.
+#
+# Setup:
+#
+# The test network consits of 3 nodes. 2 of them are producer nodes, with
+# producera and producerb. Configure --max-reversible-blocks but
+# disable --production-pause-vote-timeout-ms to avoid interfernce.
+#
+# node0: Enables block production for producera and has the finalizer key with
+#        description of producera. Has vote-threads enabled. Connect to producerbNode.
+# producerbNode: Enables block production for producerb. Has vote-threads enabled.
+#        Connect to finalizerbNode and node0.
+# finalizerbNode: Has the finalizer key with description of producerb.
+#        Has vote-threads enabled. Connect to producerbNode.
+#
+# Test cases:
+#
+# 1. Bring down finalizerbNode. producerbNode and node0 should automatically
+#    pause production due to the number of reversible blocks keeping growing
+#    because not receiving votes from finalizerbNode and LIB stopping advancings.
+# 2. Bring down node0 and restart it with a larger --max-reversible-blocks.
+#    node0 should resume production until the new limit is reached.
+# 3. Restart finalizerbNode. producerbNode and node0 should automatically unpause
+#    production because the number of reversible blocks decreases due to LIB
+#    advances on both nodes.
+####################################################################################
+
+Print=Utils.Print
+errorExit=Utils.errorExit
+
+args=TestHelper.parse_args({"-d","--keep-logs","--dump-error-details","-v","--leave-running","--unshared"})
+delay=args.d
+debug=args.v
+dumpErrorDetails=args.dump_error_details
+pnodes=2 # number of producing nodes. producers are defproducera and defproducerb
+totalNodes=pnodes + 1 # 1 is for the finalizer node of defproducerb
+prodCount=1 # number of producers per producing node
+
+Utils.Debug=debug
+testSuccessful=False
+
+cluster=Cluster(unshared=args.unshared, keepRunning=args.leave_running, keepLogs=args.keep_logs)
+walletMgr=WalletMgr(True, keepRunning=args.leave_running, keepLogs=args.keep_logs)
+
+try:
+    ####################### setting up  ######################
+    TestHelper.printSystemInfo("BEGIN")
+
+    cluster.setWalletMgr(walletMgr)
+
+    Print(f'producing nodes: {pnodes}, totalNodes: {totalNodes}, prodCount: {prodCount}, delay between nodes launch: {delay} second{"s" if delay != 1 else ""}')
+
+    # Set --max-reversible-blocks to a small number so it is exceeded quickly
+    # but it needs to be big enough for transitioning to Savanna..
+    # Disable production-pause-vote-timeout so that production pause is only
+    # caused by max-reversible-blocks reached.
+    extraNodeosArgs="--max-reversible-blocks 48 --production-pause-vote-timeout-ms 0"
```

Comment:
Instead maybe shutdown/relaunch after transition; then you don't have to guess on a number of blocks and the test runs quicker.